qos-service-impl/src/test/java/com/palantir/atlasdb/qos/ratelimit/CassandraMetricMeasurementsLoaderTest.java

```diff
@@ -69,21 +67,6 @@ public void setup() {
                 .thenReturn(METRIC_VALUE_2);
     }
 
-    @Test
-    public void metricsLoaderLoadsMetricsEveryFiveSeconds() {
-        scheduledExecutorService.tick(1, TimeUnit.SECONDS);
-        verify(HEALTH_METRIC_SUPPLIER, times(1)).get();
-
-        scheduledExecutorService.tick(4, TimeUnit.SECONDS);
-        verify(HEALTH_METRIC_SUPPLIER, times(2)).get();
-
-        scheduledExecutorService.tick(5, TimeUnit.SECONDS);
-        verify(HEALTH_METRIC_SUPPLIER, times(3)).get();
-
-        scheduledExecutorService.tick(5, TimeUnit.SECONDS);
-        verify(HEALTH_METRIC_SUPPLIER, times(4)).get();
-    }
-
```

Comment:
It seems reasonable to have _a test_ that checks that with the passage of time we hit the supplier again (this specific one might be way too strong) though - seems like current set of tests would admit a solution that caches a loaded set of values and never hits the supplier again, unless I'm missing something

Reply:
I deleted it because the test seemed pretty wild, how does it know that it has to be 5 seconds that passes? It doesn't even check that if you call it smaller than 5 seconds you'd get the same thing/or the count wouldn't go up. 

Ideally the impl would use something like Guava `AbstractScheduledService` and we just test the `runOneIteration` method.

Reply:
The test as written is not good (for the reasons you mention), but I think the concern that the current set of tests admits a clearly deficient solution still holds (hence 'a test' as opposed to 'this test').